website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -101,17 +186,96 @@ You should now be able to register your plugin into the vault catalog. To do
+   confusing for authors. For instance: passwords were handled in 3 different ways.
+   `CreateUser` generated a password and returned it, `SetCredentials` receives a password
+   via a configuration struct and returns it, and `RotateRootCredentials` generated a
+   password and was expected to return an updated copy of its entire configuration
+   with the new password.
+4. The `SetCredentials` and `RotateRootCredentials` used for static credential rotation,
+   and rotating the root user's credentials respectively were essentially the same operation:
+   change a user's password. The only practical difference was which user it was referring
+   to. This was especially evident when `SetCredentials` was used when rotating root
+   credentials (unless static credential rotation wasn't supported by the plugin in question).
+
+The new interface is roughly modeled after a [gRPC](https://grpc.io/) interface. It has improved
+future compatibility by not requiring changes to the interface definition to add additional data
+in the requests or responses. It also simplifies the interface by merging several into a single
+function call.
+
+### Upgrading your custom database
+
+Vault 1.6 supports both version 4 and version 5 database plugins. The support for version 4
+plugins will be removed in a future release. If you are using a version 4 custom database
+plugin, the following are basic instructions for upgrading to version 5.
+
+-> In version 4, password generation was the responsibility of the plugin via the
+   `GenerateCredentials` function. This is no longer the case with version 5. Vault is
+   responsible for generating passwords and passing them to the plugin via
+   `NewUserRequest.Password` and `UpdateUserRequest.Password.NewPassword`.
+
+1. Change the import path from `github.com/hashicorp/vault/sdk/database/dbplugin` to
+   `github.com/hashicorp/vault/sdk/database/dbplugin/v5`. The package name is the same, so any
+   references to `dbplugin` can remain as long as those symbols exist within the new package
+   (such as the `Serve` function).
+2. An easy way to see what functions that need to be implemented is to put the following as a
+   global variable within your package: `var _ dbplugin.Database = (*MyDatabase)(nil)`. This
+   will fail to compile if the `MyDatabase` type does not adhere to the `dbplugin.Database` interface.
+3. Update the initialization function. This step can be a little confusing since there are
+   three functions to keep in mind: In the version 4 interface, there was `Initialize` and
+   `Init`. The `Initialize` function was replaced with the `Init` function and was not used
+   some time before Vault 1.6. It was kept in the interface originally for backwards
+   compatibility. The version 5 interface has an `Initialize` function that will need to replace
+   the `Init` function. The two arguments passed into `Init` (`config map[string]interface{}`
```

Comment:
This block is quite hard to follow. I think some of the historical context could be pulled out into the background section above to leave this bullet as just the imperative instructions required to migrate.

Reply:
Updated. Let me know how it reads.

Reply:
Way clearer now 💯  nice job